extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -56,249 +123,144 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 				.withAdditionalProperties(propertyBag).withConnectionInfo(connectionInfo).send();
 
 			// list databases based on connection profile selected
-			connectionURI = await vscodeMssqlApi.connect(connectionInfo);
-			listDatabases = await vscodeMssqlApi.listDatabases(connectionURI);
-			const selectedDatabase = (await vscode.window.showQuickPick(listDatabases, {
-				canPickMany: false,
-				title: constants.selectDatabase,
-				ignoreFocusOut: true
-			}));
-
+			telemetryStep = CreateAzureFunctionStep.getDatabase;
+			let selectedDatabase = await azureFunctionsUtils.getDatabase(connectionInfo);
 			if (!selectedDatabase) {
 				// User cancelled
-				return;
+				return undefined;
 			}
 			connectionInfo.database = selectedDatabase;
 
 			// prompt user for object name to create function from
+			telemetryStep = CreateAzureFunctionStep.getObjectName;
 			objectName = await azureFunctionsUtils.promptForObjectName(selectedBinding.type);
 			if (!objectName) {
 				// user cancelled
 				return;
 			}
-
-		} catch (e) {
-			propertyBag.quickPickStep = quickPickStep;
-			exitReason = 'error';
-			TelemetryReporter.createErrorEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.exitCreateAzureFunctionQuickpick, undefined, utils.getErrorType(e))
-				.withAdditionalProperties(propertyBag).send();
-			return;
-		}
-	} else {
-		quickPickStep = 'launchFromTable';
-		connectionInfo = node.connectionInfo;
-		// set the database containing the selected table so it can be used
-		// for the initial catalog property of the connection string
-		let newNode: ITreeNodeInfo = node;
-		while (newNode) {
-			if (newNode.nodeType === 'Database') {
-				connectionInfo.database = newNode.metadata.name;
-				break;
-			} else {
-				newNode = newNode.parentNode;
+		} else {
+			// if user selects table in tree view we use connection info from Object Explorer node
+			telemetryStep = CreateAzureFunctionStep.launchFromTable;
+			connectionInfo = node.connectionInfo;
+			// set the database containing the selected table so it can be used
+			// for the initial catalog property of the connection string
+			let newNode: ITreeNodeInfo = node;
+			while (newNode) {
+				if (newNode.nodeType === 'Database') {
+					connectionInfo.database = newNode.metadata.name;
+					break;
+				} else {
+					newNode = newNode.parentNode;
+				}
 			}
+
+			objectName = utils.generateQuotedFullName(node.metadata.schema, node.metadata.name);
 		}
-		// Ask binding type for promptObjectName
-		quickPickStep = 'getBindingType';
-		let selectedBinding = await azureFunctionsUtils.promptForBindingType();
+		telemetryStep = CreateAzureFunctionStep.getConnectionString;
```

Comment:
why do we need telemetryStep? It seems that you are reassigning it's value without sending it as telemetry

Reply:
So currently I have this so that if an error occurs at any point or if the user cancels a certain step within the flow then in the finally block we would send where they left off from.